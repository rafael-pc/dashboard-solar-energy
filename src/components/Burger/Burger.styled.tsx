import styled from "styled-components";
import { theme } from "../../styles";

export const Burger = styled.button<{ open: boolean }>`
  position: absolute;
  top: 9px;
  right: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  padding: 0;
  z-index: 10;

  span {
    width: 2rem;
    height: 0.25rem;
    background: ${theme.colors.neutral_100};
    border-radius: ${theme.border.radius.m};
    transition: all 0.3s linear;
    position: relative;
    transform-origin: 1px;

    &:first-child {
      transform: ${({ open }) => (open ? "rotate(45deg)" : "rotate(0)")};
    }

    &:nth-child(2) {
      opacity: ${({ open }) => (open ? "0" : "1")};
      transform: ${({ open }) => (open ? "translateX(20px)" : "translateX(0)")};
    }

    &:nth-child(3) {
      transform: ${({ open }) => (open ? "rotate(-45deg)" : "rotate(0)")};
    }
  }
`;
