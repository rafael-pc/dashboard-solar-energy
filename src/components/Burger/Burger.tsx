import React from "react";
import * as S from "./Burger.styled";

interface BurgerProps {
  open: boolean;
  setOpen: (open: boolean) => void;
}

const Burger: React.FC<BurgerProps> = ({ open, setOpen, ...props }) => {
  const isExpanded = open ? true : false;

  return (
    <S.Burger
      aria-label="Toggle menu"
      aria-expanded={isExpanded}
      open={open}
      onClick={() => setOpen(!open)}
      {...props}
    >
      <span />
      <span />
      <span />
    </S.Burger>
  );
};

export default Burger;
