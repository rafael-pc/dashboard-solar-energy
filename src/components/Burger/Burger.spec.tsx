import { render, fireEvent } from "@testing-library/react";
import Burger from "./Burger";

describe("Burger", () => {
  it("renders correctly when closed", () => {
    const setOpen = jest.fn();
    const { container } = render(<Burger isOpen={false} setOpen={setOpen} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly when open", () => {
    const setOpen = jest.fn();
    const { container } = render(<Burger isOpen={true} setOpen={setOpen} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls setOpen with the opposite value when clicked", () => {
    const setOpen = jest.fn();
    const { getByLabelText } = render(
      <Burger isOpen={false} setOpen={setOpen} />
    );
    fireEvent.click(getByLabelText("Toggle menu"));
    expect(setOpen).toHaveBeenCalledWith(true);
  });
});
