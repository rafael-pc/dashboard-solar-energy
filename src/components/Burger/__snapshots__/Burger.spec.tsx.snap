// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Burger renders correctly when closed 1`] = `
<button
  aria-expanded="false"
  aria-label="Toggle menu"
  class="sc-bcPKhP cQSnCq"
>
  <span />
  <span />
  <span />
</button>
`;

exports[`Burger renders correctly when open 1`] = `
<button
  aria-expanded="true"
  aria-label="Toggle menu"
  class="sc-bcPKhP FIjTY"
  open=""
>
  <span />
  <span />
  <span />
</button>
`;
